@xen-orchestra/rest-api/src/abstract-classes/xo-controller.mts

```diff
@@ -0,0 +1,47 @@
+import * as CM from 'complex-matcher'
+import type { NonXapiXoRecord } from '@vates/types/xo'
+
+import { BaseController } from './base-controller.mjs'
+
+import { RestApi } from '../rest-api/rest-api.mjs'
+
+export abstract class XoController<T extends NonXapiXoRecord> extends BaseController<T, false> {
+  abstract _abstractGetObjects(): Promise<T[]>
+  abstract _abstractGetObject(id: T['id']): Promise<T>
```

Comment:
I think theses should be named _getObject / _getObject. It seems strang to have an implemented method called _abstractSomething